modal/queue.py

```diff
@@ -23,11 +23,31 @@ class _Queue(_Object, type_prefix="qu"):
 
     The queue can contain any object serializable by `cloudpickle`, including Modal objects.
 
-    **Lifetime of a queue and its contents**
+    By default, the `Queue` object acts as a single FIFO queue which supports puts and gets (blocking and non-blocking).
 
-    A `Queue`'s lifetime matches the lifetime of the app it's attached to, but the contents expire after 30 days.
-    Because of this, `Queues`s are best used for communication between active functions and not relied on for
-    persistent storage. On app completion or after stopping an app any associated `Queue` objects are cleaned up.
+    **Queue partitions (beta)**
+
+    Specifying partition keys gives access to other independent FIFO partitions within the same `Queue` object.
+    Across any two partitions, puts and gets are completely independent. For example, a put in one partition does not affect
+    a get in any other partition.
+
+    When no partition key is specified (by default), puts and gets will operate on a default partition. This default partition
+    is also isolated from all other partitions. Please see the Usage section below for an example using partitions.
+
+    **Lifetime of a queue and its partitions**
+
+    By default, each partition is cleared 24 hours after the last `put` operation. A lower TTL can be specified by the `partition_ttl`
+    argument in the `put` or `put_many` methods. Each partition's expiry is handled independently.
+
+    As such, `Queue`s are best used for communication between active functions and not relied on for persistent storage.
+
+    On app completion or after stopping an app any associated `Queue` objects are cleaned up. All its partitions will be cleared.
+
+    **Limits**
+
+    A single `Queue` can contain up to 100,000 partitions, each with up to 5,000 items. Each item can be up to 256 KiB.
```

Comment:
Should we also document the total memory limit?

Reply:
Could also mention what happens if you exceed these limits (e.g. if there's a specific error class that could be caught).